packages/tpetra/core/test/ImportExport2/AsyncTransfer_UnitTests.cpp

```diff
@@ -693,6 +780,246 @@ namespace {
   }
 
 
+  template <typename Packet, typename LO, typename GO>
+  class CheckDefaultTransferStatus {
+  private:
+    using DistObjectRCP = RCP<DistObject<Packet, LO, GO>>;
+
+  public:
+    CheckDefaultTransferStatus(FancyOStream& o, bool& s)
+      : out(o),
+        success(s)
+    { }
+
+    void operator()(DistObjectRCP source, DistObjectRCP target) const {
+      TEST_ASSERT(target->transferArrived());
+    }
+
+  private:
+    FancyOStream& out;
+    bool& success;
+  };
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL( TransferArrived, MultiVector_defaultTrue, Scalar, LO, GO )
```

Comment:
For each test, having a one-sentence comment describing the feature being exercised would be helpful.  I spent a bit of time trying to figure out why this test didn't actually do an import/export, before realizing that it was just checking that transferArrived is true when an import/export hasn't been initiated.    So for all tests, a comment like the following would be helpful:
```
// Test that transferArrived is true when no multivector transfer has been initiated 
```
or 
```
// Test that multivector transfer works correctly with target multivector completely on rank 0
```

Reply:
I tend to view a need for comments as an indication that I've failed to write code that's clear enough to stand on its own (as is obviously true in this case) -- so my first preference is to rewrite the code to be more clear and obvious about its intent.  Is there another name for this functor or for the test/fixtures that would make the intent of the test more clear to you?  Or should I just admit failure and add the comments?